src/sampling.jl

```diff
@@ -518,7 +636,9 @@ function sample!(a::AbstractArray, wv::WeightVec, x::AbstractArray;
             end
         end
     else
-        naive_wsample_norep!(a, wv, x)
+        k <= n || error("Cannot draw more samples without replacement.")
```

Comment:
Here too, printing `k` and `n` could be useful to understand the reasons for the error.